src/number_theory/temporary_file.lean

```diff
@@ -0,0 +1,267 @@
+import ring_theory.adjoin_root
+import field_theory.minpoly
+import data.polynomial.div
+import ring_theory.integrally_closed
+import field_theory.splitting_field
+
+open polynomial localization
+
+
+open_locale polynomial
+
+variables {R S : Type*} [comm_ring R] [comm_ring S]
+
+section move_me
+/- The results in this section are useful (they are direct generalisations of results in
+  `gauss_lemma.lean` but aren't necessary for this PR so will be shortly moved to another one) -/
+
+theorem monic_map_of_monic [nontrivial S] (f : R →+* S) {p : R[X]} (hp : p.monic):
+  monic (p.map f) :=
+begin
+  have : nat_degree p = nat_degree (p.map f),
+  { refine le_antisymm (le_nat_degree_of_ne_zero _) (nat_degree_map_le _ _),
+    rw [polynomial.coeff_map, coeff_nat_degree, monic.def.mp hp, ne.def,
+      ← ring_hom.codomain_trivial_iff_map_one_eq_zero],
+    exact zero_ne_one },
+  rw [monic.def, leading_coeff, coeff_map, ← this, ← leading_coeff, monic.def.mp hp, map_one],
+end
+
+theorem monic_map_iff_of_injective [nontrivial S] (f : R →+* S) (hf : function.injective f)
```

Comment:
I'd call this `function.injective.monic_map_iff`, for dot notation. And also I'd rename `polynomial.monic_of_injective` to maybe `function.injective.monic_of_map`, but probably out of scope